### PR TITLE
Added option to add LOCK MODE WAIT in the connection

### PR DESCRIPTION
**Pull request by shahha**

### CODE REVIEW

**Comment by RobbieClarken** [comment]
LGTM. If you wanted to avoid the three levels or nested `if` statements it could be rewritten as:

```python
if 'LOCK_MODE_WAIT' in conn_params['OPTIONS']:
    wait_parameter = conn_params['OPTIONS']['LOCK_MODE_WAIT']
    if wait_parameter == 0:
        sql = 'SET LOCK MODE TO NOT WAIT'
    elif wait_parameter == -1:
        sql = 'SET LOCK MODE TO WAIT'
    else:
        sql = 'SET LOCK MODE TO WAIT {}'.format(wait_parameter)
    self.connection.cursor().execute(sql)
```

File: README.rst [hunk 1]
```diff
@@ -57,13 +57,31 @@ Django’s settings.py require the following format to connect to an Informix da
...
+ISOLATION_LEVEL
+    This will set database isolation level at connection level
+    Possible values:
+        READ_COMMITED
+        READ_UNCOMMITTED
+        SERIALIZABLE
+
+LOCK_MODE_WAIT
+    This will set database LOCK MODE WAIT at connection level
+    Application can use this property to override the default server process for accessing a locked row or table.
```
Comment (richard-reece) [inline README.rst]: This line is >80 chars, which is longer than most terminals, which will make it harder to read. Please reformat.

File: README.rst [hunk 2]
```diff
@@ -57,13 +57,31 @@ Django’s settings.py require the following format to connect to an Informix da
         'PASSWORD': 'passw0rd',
         'OPTIONS': {
             'DRIVER': '/path/to/iclit09b.so'. # Or iclit09b.dylib on macOS
+            'CPTIMEOUT': 120,
```
Comment (richard-reece) [inline README.rst]: I know it's not related to this patch, but could you please add some words explaining what a "CPTIMEOUT" is?

File: django_informixdb/base.py [hunk 3]
```diff
@@ -244,6 +244,23 @@ def get_new_connection(self, conn_params):
...
+        # Possible values:
+        #    -1 - WAIT until the lock is released.
+        #    0 - DO NOT WAIT, end the operation, and return with error.
+        #    nn - WAIT for nn seconds for the lock to be released.
+        if 'LOCK_MODE_WAIT' in conn_params['OPTIONS']:
+            if conn_params['OPTIONS']['LOCK_MODE_WAIT']:
+                wait_parameter = conn_params['OPTIONS']['LOCK_MODE_WAIT']
+                if wait_parameter == -1:
+                    wait_parameter = ''
+                sql = 'SET LOCK MODE TO WAIT ' + str(wait_parameter)
```
Comment (richard-reece) [inline django_informixdb/base.py]: A safer line here would be:

    sql = 'SET LOCK MODE TO WAIT %d' % wait_parameter

as that will also type-check the parameter to ensure it's an integer, hopefully preventing accidental weird configs.

File: README.rst (hunk 2 above)
Comment (shahha) [reply README.rst]: Added.

File: django_informixdb/base.py (hunk 3 above)
Comment (shahha) [reply django_informixdb/base.py]: This will not handle all possible situation. So keeping it as is.

**Pull request merged**